app/screens/permalink/permalink.js

```diff
@@ -86,10 +86,45 @@ export default class Permalink extends PureComponent {
         intl: intlShape.isRequired,
     };
 
+    static getDerivedStateFromProps(nextProps, prevState) {
+        const newState = {};
+        if (nextProps.focusedPostId !== prevState.focusedPostIdState) {
+            newState.focusedPostIdState = nextProps.focusedPostId;
+        }
+
+        if (nextProps.channelId && nextProps.channelId !== prevState.channelIdState) {
+            newState.channelIdState = nextProps.channelId;
+        }
+
+        if (nextProps.channelName && nextProps.channelName !== prevState.channelNameState) {
+            newState.channelNameState = nextProps.channelName;
+        }
+
+        if (nextProps.postIds.length > 0 && nextProps.postIds !== prevState.postIdsState) {
+            newState.postIdsState = nextProps.postIds;
+        }
+
+        if (nextProps.focusedPostId !== prevState.focusedPostIdState) {
+            let loading = true;
+            if (nextProps.postIds && nextProps.postIds.length >= 10) {
+                loading = false;
+            }
+
+            newState.loading = loading;
+        }
+
+        return newState;
```

Comment:
if the state does not change you should return null here

Reply:
Looks like empty object has the same effect as null on `getDerivedStateFromProps` when no need to change the state since the test I put in place in matching states remain unchanged. Anyway, I added check that detects empty object and return null instead.